### PR TITLE
Allow to install Agent integration

### PR DESCRIPTION
**Pull request by ogaca-dd**
What does this PR do?
---------------------

This PR adds the support for integrations. It also introduces `WithTelemetry` which provides the Agent telemetry (package `telemetry` and `go expvar`).

There is an issue when updating from `_, err = agent.NewInstaller(vm, agent.WithTelemetry())` to ` _, err = agent.NewInstaller(vm)`. Indeed, even with `pulumi.DeleteBeforeReplace(true)` on `restart-agent` resource, `restart-agent` is run before the integration files were removed. Any suggestions?

Which scenarios this will impact?
-------------------

Motivation
----------

Additional Notes
----------------

### CODE REVIEW

File: datadog/agent/install.go [hunk 1]
```diff
@@ -68,6 +69,20 @@ func NewInstaller(vm *vm.UnixLikeVM, options ...func(*params) error) (*Installer
 	return &Installer{dependsOn: lastCommand}, err
 }
 
+func addTelemetry(options []func(*params) error) []func(*params) error {
+	config := `
+instances:
```
Comment (pducolin) [inline datadog/agent/install.go]: **💭 thought**
You can embed this - advantage is yaml formatting.

**Review by pducolin — APPROVED** [review]

**Comment by pducolin** [comment]
> restart-agent is run before the integration files were removed

Is there anything else that changed, and that the `restart-agent` depends on ?

File: datadog/agent/install.go [hunk 2]
```diff
@@ -65,7 +58,7 @@ func NewInstaller(vm *vm.UnixLikeVM, options ...func(*params) error) (*Installer
 	// When the file content has changed, make sure the Agent is restarted.
 	serviceManager := os.GetServiceManager()
 	for _, cmd := range serviceManager.RestartAgentCmd() {
-		restartAgentRes := commonNamer.ResourceName("restart-agent", utils.StrHash(cmd, params.agentConfig, utils.StrHash(hash)))
+		restartAgentRes := commonNamer.ResourceName("restart-agent", utils.StrHash(cmd, configHash, integsHash))
```
[the remark this answers is not in the document]
Comment (ogaca-dd) [reply datadog/agent/install.go]: What do you mean by a range of restart commands?

File: datadog/agent/install.go (hunk 1 above)
Comment (ogaca-dd) [reply datadog/agent/install.go]: I think the inline content is small enough to be kept in the code (for a better locality).

**Comment by ogaca-dd** [comment]
> Is there anything else that changed, and that the restart-agent depends on ?
As far as I know no.

File: datadog/agent/install.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (pducolin) [reply datadog/agent/install.go]: ~range~ a slice, why do we have to iterate through `serviceManager.RestartAgentCmd()` ?

**Pull request merged**